ledger-tool/src/bigtable.rs

```diff
@@ -336,6 +338,250 @@ pub async fn transaction_history(
+                    return Err(Box::new(err));
+                }
+            }
+        } else {
+            match solana_storage_bigtable::LedgerStorage::new_with_config(
+                solana_storage_bigtable::LedgerStorageConfig {
+                    read_only: true,
+                    timeout: None,
+                    credential_type: CredentialType::Filepath(Some(
+                        args.destination_credential_path.unwrap(),
+                    )),
+                    instance_name: args.destination_instance_name,
+                    app_profile_id: args.destination_app_profile_id,
+                },
+            )
+            .await
+            {
+                Ok(bigtable) => bigtable,
+                Err(err) => {
+                    error!("Failed to connect to destination bigtable: {:?}", err);
+                    return Err(Box::new(err));
+                }
+            }
+        }
+    };
+
+    let (s, r) = unbounded::<u64>();
+    for i in from_slot..to_slot {
+        s.send(i).unwrap();
+    }
+
+    let workers = min(to_slot - from_slot, num_cpus::get().try_into().unwrap());
+    debug!("worker num: {}", workers);
+
+    let success_slots = Arc::new(Mutex::new(vec![]));
+    let skip_slots = Arc::new(Mutex::new(vec![]));
+    let block_not_found_slots = Arc::new(Mutex::new(vec![]));
+    let failed_slots = Arc::new(Mutex::new(vec![]));
+
+    let tasks = (0..workers)
+        .map(|i| {
+            let r = r.clone();
+            let source_bigtable_clone = source_bigtable.clone();
+            let destination_bigtable_clone = destination_bigtable.clone();
+
+            let success_slots_clone = Arc::clone(&success_slots);
+            let skip_slots_clone = Arc::clone(&skip_slots);
+            let block_not_found_slots_clone = Arc::clone(&block_not_found_slots);
+            let failed_slots_clone = Arc::clone(&failed_slots);
+            tokio::spawn(async move {
+                while let Ok(slot) = r.try_recv() {
+                    debug!("worker {}: received slot {}", i, slot);
+
+                    if !args.force {
+                        match destination_bigtable_clone.get_confirmed_block(slot).await {
+                            Ok(_) => {
+                                skip_slots_clone.lock().unwrap().push(slot);
+                                continue;
+                            }
+                            Err(solana_storage_bigtable::Error::BlockNotFound(_)) => {}
+                            Err(err) => {
+                                debug!("failed to get confirmed block from destination, slot: {}, err: {}", slot, err);
+                                failed_slots_clone.lock().unwrap().push(slot);
+                                continue;
+                            }
+                        };
+                    }
+
+                    let confirmed_block =
+                        match source_bigtable_clone.get_confirmed_block(slot).await {
```

Comment:
lame that there's no way to copy directly between BigTable instances rather than pull all source blocks down to the `solana-ledger-tool` host and push them back up. :face_with_head_bandage:

Reply:
ah. i was hoping there'd be a way to just directly copy those columns(?) as well.  oh well